### PR TITLE
[EOC-24] add react router and setup routing

### PR DESCRIPTION
**Pull request by alemikolo**

### CODE REVIEW

File: client/src/app/index.jsx [hunk 1]
```diff
@@ -1,17 +1,36 @@
...
 
-ReactDOM.render(
+const store = configureStore();
+const eoc = (
   <Provider store={store}>
-    <Main />
-  </Provider>,
-  document.getElementById('root')
+    <Router>
+      <div>
```
Comment (marekrozmus) [inline client/src/app/index.jsx]: remove unneeded `div`

File: client/src/modules/cohort/components/Cohort/Cohort.jsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const Cohort = ({
+  match: {
+    params: { id }
+  }
+}) => <div style={{ padding: '100px' }}>{`Cohort of id: ${id}`}</div>;
```
Comment (marekrozmus) [inline client/src/modules/cohort/components/Cohort/Cohort.jsx]: please move to scss file

File: client/src/modules/dashboard/components/Dashboard/Dashboard.jsx [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import React from 'react';
+
+const Dashboard = () => <div style={{ padding: '100px' }}>Dashboard</div>;
```
Comment (marekrozmus) [inline client/src/modules/dashboard/components/Dashboard/Dashboard.jsx]: please remove or move to scss file

File: client/src/modules/main/components/Main/Main.jsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+import Auth from 'modules/legacy/Auth';
+
+const Main = ({
+  match: {
+    params: { id }
+  }
+}) => (
+  <div>
+    <div style={{ padding: '100px' }}>Main aplication page</div>
```
Comment (marekrozmus) [inline client/src/modules/main/components/Main/Main.jsx]: see previous comment

File: client/src/app/index.jsx [hunk 5]
```diff
@@ -1,17 +1,34 @@
...
+const eoc = (
   <Provider store={store}>
-    <Main />
-  </Provider>,
-  document.getElementById('root')
+    <Router>
+      <Switch>
+        <Route exact path="/" component={Main} />
+        <Route path="/dashboard" component={Dashboard} />
+        <Route path="/cohort/:id(\w+)" component={Cohort} />
```
Comment (adamklepacz) [inline client/src/app/index.jsx]: what does (\w+) mean in path param?

File: client/src/modules/shopping-list/components/ShoppingList/ShoppingList.jsx [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const ShoppingList = ({
+  match: {
+    params: { id }
+  }
+}) => <div>{`Cohort of id: ${id}`}</div>;
+
```
Comment (adamklepacz) [inline client/src/modules/shopping-list/components/ShoppingList/ShoppingList.jsx]: Shooping list of id

File: client/src/app/index.jsx (hunk 5 above)
Comment (alemikolo) [reply client/src/app/index.jsx]: it's regexp which validates id param, accepted values: alphanumeric characters and _, if the validation fails, the main page will be load: <Redirect to="/" />

**Pull request closed without merging**